### DISCUSSION TITLE
Duda uso de memoria

### DISCUSSION THREAD

**Discussion by MaxAl100** [IIC2133-PUC/2024-1#64] [Tarea 1] [answered]
Tengo una duda respecto al uso de memoria, ya que se menciona que no debemos usar más de 1 GB de RAM, pero al no saber el tamaño del input no sé si será necesario implementar algoritmos in-place o no. Esto tenemos que verlo experimentando o es seguro que, a no ser que programemos muy mal, no nos pasaremos de este límite de memoria?

**Comment by sachondo7** [comment, marked as answer]
Está tarea sobre todo juega mucho con la complejidad de los algoritmos, si respetas lo que pide el enunciado no deberías pasarte del giga de ram. 
Si quieres o no hacer los algoritmos in place es decisión tuya, solo asegúrate de ir liberando la memoria a medida que ya no la vayas necesitando.

**Reply by MaxAl100** [reply]
Yap, muchas gracias! Eso pensaba, pero quería asegurarme de que era así 😃
